### PR TITLE
Added error messages

### PR DESCRIPTION
**Pull request by NikolaJelic**
Added: 
-  optional error messages which the user can turn on or off
-  excessive white space trimming
-  lines starting with `#` which can be considered comments are ignored
-  lines with empty keys are ignored

### CODE REVIEW

**Comment by hentai-chan** [comment]
I'll submit my review tomorrow, there are a few edge cases that I want to test

File: include/pini/util.hpp [hunk 1]
```diff
@@ -10,10 +10,11 @@
...
 
 // extract key-value pairs from given vector of strings and insert them in an
 // unordered_map
-std::unordered_map<std::string, std::string>
-insert_pairs(std::vector<std::string> file_lines);
-} // namespace util
\ No newline at end of file
+std::unordered_map<std::string, std::string> insert_pairs(std::vector<std::string> file_lines);
+
+std::string trim_whitespace(std::string const& str);
```
Comment (karnkaul) [inline include/pini/util.hpp]: Why not use string_views for all such operations and avoid pointless allocations etc in the middle of substringing?

File: src/pini.cpp [hunk 2]
```diff
@@ -65,7 +81,20 @@ std::int32_t pini::get_int32(std::string const& key, std::uint32_t def) const {
 
 std::string_view pini::get_string(std::string const& key) const {
 	auto it = key_value_pairs.find(key);
-	if (it == key_value_pairs.end()) { return {}; }
+	if (it == key_value_pairs.end()) {
+		if (pn::pini::on_msg_t) {
```
Comment (karnkaul) [inline src/pini.cpp:77]: Are you ok with repeating this pattern every time you need to log stuff? There are ways to make it more succinct.

File: src/util.cpp [hunk 3]
```diff
@@ -9,20 +12,43 @@ std::vector<std::string> get_lines(std::filesystem::path const& filename) {
...
+			(*pn::pini::on_msg_t)(str.str(), pn::pini::severity::error);
+		}
 	}
 	return ret;
 }
 
 std::unordered_map<std::string, std::string> insert_pairs(std::vector<std::string> file_lines) {
 	std::unordered_map<std::string, std::string> key_value_pairs;
-
+	std::size_t line_number = 0;
```
Comment (karnkaul) [inline src/util.cpp:27]: Nice, you used size_t and not int

File: src/util.cpp [hunk 4]
```diff
@@ -9,20 +12,43 @@ std::vector<std::string> get_lines(std::filesystem::path const& filename) {
...
+				std::string const value = trim_whitespace(str.substr(delimeter_pos + 1));
+				if (key[0] != '#') { key_value_pairs.insert({key, value}); }
+			}
 		}
 	}
 	return key_value_pairs;
 }
 
+std::string trim_whitespace(std::string const& str) {
+
```
Comment (karnkaul) [inline src/util.cpp]: Why these random blank lines

File: src/util.cpp [hunk 5]
```diff
@@ -9,20 +12,43 @@ std::vector<std::string> get_lines(std::filesystem::path const& filename) {
...
+			} else {
+				std::string const value = trim_whitespace(str.substr(delimeter_pos + 1));
+				if (key[0] != '#') { key_value_pairs.insert({key, value}); }
+			}
 		}
 	}
 	return key_value_pairs;
 }
 
+std::string trim_whitespace(std::string const& str) {
```
Comment (karnkaul) [inline src/util.cpp]: Again, these can be string views, you aren't modifying any chars, and the returned string is a contiguous substring of the passed one

File: test/pini_test.cpp [hunk 6]
```diff
@@ -5,8 +5,10 @@
 int main() {
 	pn::pini pin;
 
-	std::string_view raw_input{"a=5\nb=10\n23=c"};
+	std::string_view raw_input{"= 5\n#b=10\n c  = 23"};
 	// std::filesystem::path filename{"/test/test.ini"};
 	if (!pin.load_text(raw_input)) { return 1; }
-	if (pin.get_int32("a") != 5) { return 1; }
```
Comment (karnkaul) [inline test/pini_test.cpp:11]: What's wrong with this line?

File: test/pini_test.cpp [hunk 7]
```diff
@@ -5,8 +5,10 @@
...
+	std::cout << pin.get_int32("a");
+	std::cout << pin.get_int32("b");
+	std::cout << pin.get_int32("c");
```
Comment (karnkaul) [inline test/pini_test.cpp:13]: I'd put some more failure cases instead of just printing stuff: you won't see these logs via ctest / through CI etc.

**Review by karnkaul — APPROVED** [review]
